### ISSUE TITLE
PV model adjustments to operating voltage are hard to follow

### ISSUE THREAD

**Issue by cpaulgilman** [NREL/SAM#146]
Some PV systems use power electronics either in the inverter or separate power optimizers to adjust the power output of the array by moving the array operating voltage along the its power curve at the given incident irradiance. SAM's PV models assume that the array operates at its maximum power point.

SAM's inverter models account for power loss due to voltage clipping that occurs when the operating voltage falls outside of the inverter's rated MPPT window. When that happens the inverter input DC voltage is set to either the minimum or maximum MPPT voltage rating so that the inverter output power is based on the adjusted input voltage, and the associated power loss can be accounted for.

SAM's Detailed Photovoltaic model allows the user to choose between different module models. The CEC Performance Model with Module Database,  CEC Performance Model with User Entered Specifications, and IEC Single Diode Model are single diode equivalent models that can calculate power anywhere along the power curve. The Simple Efficiency Model and Sandia PV Array Performance Model with Module Database are point-value models that can only calculate power at certain points on the curve, and are not suitable for modeling systems that can adjust the array output power.

The single diode models allow SAM to more realistically model array voltage, but in order to implement that capability, we would have to remove the point-value model options.

**Comment by janinefreeman** [comment]
@adriesse I believe the proposal is to allow SAM to adjust the operating voltage of a string when the inverter clips the module outside of its MPPT operating range. So we would re-run the module model at an off-MPP voltage after calculating that at the module MPP, the string voltage would be outside the inverter MPPT operating range. Our latest investigations, however, have shown that this is in fact what SAM is doing for any of the single diode models (CEC with user specifications, CEC from datasheet, and 61853 model)- I added this functionality when I re-wrote that section of the model to allow for multiple MPPT inputs on an inverter. But, the output labels are unclear so it's not obvious that the model IS adjusting the operating voltage. I've asked Paul for suggestions on how to communicate this behavior more effectively.

**Comment by adriesse** [comment]
Thanks, Janine, that helps.  I do understand the general issue, but the title of this issue alarmed me.  I understand now that the title is actually wrong, and the models do not need to be changed.

**Comment by janinefreeman** [comment]
Fair enough! Updated the title to hopefully cut down on the confusion. :)

**Comment by adriesse** [comment]
Do you do the same trick (re-run the module model) when it is found that the array power exceeds the allowable inverter input power?
